ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/issues/index.tsx

```diff
@@ -18,67 +18,282 @@
  */
 
 import { createFileRoute } from '@tanstack/react-router';
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getFilteredRowModel,
+  getPaginationRowModel,
+  Row,
+  useReactTable,
+} from '@tanstack/react-table';
+import { Minus, Plus } from 'lucide-react';
+import { useMemo } from 'react';
 
+import { useIssuesServiceGetIssuesByRunId } from '@/api/queries';
 import { prefetchUseRepositoriesServiceGetOrtRunByIndex } from '@/api/queries/prefetch';
 import { useRepositoriesServiceGetOrtRunByIndexSuspense } from '@/api/queries/suspense';
+import { Issue } from '@/api/requests';
+import { DataTable } from '@/components/data-table/data-table';
+import { DataTableToolbar } from '@/components/data-table/data-table-toolbar';
+import { FilterMultiSelect } from '@/components/data-table/filter-multi-select';
 import { LoadingIndicator } from '@/components/loading-indicator';
+import { ToastError } from '@/components/toast-error';
+import { Badge } from '@/components/ui/badge';
+import { Button } from '@/components/ui/button';
 import {
   Card,
   CardContent,
   CardDescription,
   CardHeader,
   CardTitle,
 } from '@/components/ui/card';
-import { Label } from '@/components/ui/label';
+import { getIssueSeverityBackgroundColor } from '@/helpers/get-status-class';
+import { toast } from '@/lib/toast';
 import { formatTimestamp } from '@/lib/utils.ts';
+import {
+  issueSeverity,
+  issueSeveritySchema,
+  paginationSchema,
+} from '@/schemas';
+
+const defaultPageSize = 10;
+
+/**
+ * Filters table rows based on severities.
+ *
+ * @param row The data row to be evaluated.
+ * @param columnId The unique identifier of the column to filter by.
+ * @param filterValue An array of severities to filter the rows by.
+ * @returns Returns true/false based on whether the row should be
+ * included in the filtered results.
+ */
+const severityFilterFn = (
+  row: Row<Issue>,
+  columnId: string,
+  filterValue: string[]
+): boolean => {
+  // TypeScript can't infer the type, so cast it explicitly.
+  // This is safe because the columnId is always one of the
+  // severities, which themselves are strings.
+  const rowValue = row.getValue(columnId) as string;
+  return filterValue.includes(rowValue);
+};
```

Comment:
As this is only used in one column, I would move the declaration in-line to the column definition, mostly in hope to gain better type-safety.

Reply:
It didn't, though.

Reply:
Looking at it here, does it just filter the exact values? Would some of the built-in functions work like:

```typescript
    filterFn: 'equalsString',
```

Reply:
> Looking at it here, does it just filter the exact values? Would some of the built-in functions work like:
> 
> ```typescript
>     filterFn: 'equalsString',
> ``` 

No, it filters against an array of possible severities, like `[WARNING, ERROR]` and there are no built-in functions for that. There are three functions that do the exact opposite - filter an array of values against an option - but no built-in functions for filtering a value against an array of options. All the solutions I could find used `includes()`.

Reply:
Thanks, makes sense.

Reply:
You can make it more type-safe like this:

```typescript
    filterFn: (
      row,
      _columnId,
      filterValue
    ): boolean => {
      return filterValue.includes(row.original.severity);
    },
```

Reply:
That was a nice simplifying fix, thank you!